File: 205-isomorphic-strings/isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        """s_hash = defaultdict(int)
        t_hash = defaultdict(int)
        for idx in range(len(s)):
            if s[idx] not in s_hash:
                s_hash[s[idx]] = idx
            
            if t[idx] not in t_hash:
                t_hash[t[idx]] = idx
            
            if s_hash[s[idx]] != t_hash[t[idx]]:
                return False
        return True"""

        return len(set(s)) == len(set(t)) == len(set(zip(s,t)))

File: 205-isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if (s.size()!= t.size()) return false;
        unordered_map<char,char> smap, tmap;
        for(int i=0; i <s.size(); i++){
            if (smap.find(s[i]) == smap.end()) {
                smap[s[i]] = i;
            } 
            if (tmap.find(t[i])==tmap.end()) {
                tmap[t[i]] = i;  
            } 
            if(smap[s[i]] != tmap[t[i]]) return false;
        }
        return true;
    }
};